### DISCUSSION TITLE
PostgrestSingleResponse weird typing

### DISCUSSION THREAD

**Discussion by JulienLecoq** [supabase/supabase#12053] [Questions]
Applying ``.single()`` to a ``.select()`` statement returns a ``PromiseLike<PostgrestSingleResponse<myType>>``.

PostgrestSingleResponse is the following sum type:
```ts
export type PostgrestSingleResponse<T> =
  | PostgrestSingleResponseSuccess<T>
  | PostgrestResponseFailure
  ```
  
PostgrestSingleResponseSuccess has the following signature:
```ts
interface PostgrestSingleResponseSuccess<T> extends PostgrestResponseBase {
  error: undefined
  data: T
  count: number | undefined
}
```

Which doesn't make sense, why is there a count property there since we're fetching only 1 item?

**Comment by steve-chavez** [comment]
Hello,

Could open a discussion/PR on https://github.com/supabase/postgrest-js/ suggesting the improvement?
